Cargo.toml

```diff
@@ -30,18 +30,18 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-arrow = "19.0.0"
-arrow-flight = "19.0.0"
-azure_core = "0.3.0"
-azure_identity = "0.4.0"
+arrow = "20.0.0"
+arrow-flight = "20.0.0"
+azure_core = "0.4.0"
+azure_identity = "0.5.0"
```

Comment:
These have had some updates since this was written. Through testing, I've noticed some breaking changes though, so it may be best left for another PR. This may also be useful, as mentioned in another comment.

Reply:
Which do you recommend as updates and which should we keep as is? Sounds like you want to bump azure_identity at least to latest 0.6.0

Reply:
I believe we should work to update azure_identity to 0.6.0. 
The breaking change should be relatively easy to fix. 0.5.0 changed exchange() to now require an HTTP client. It also changes its oauth2 crate dependency.
The changelogs for azure_identity (along with the other Azure SDK for Rust crates) can be found [here](https://github.com/Azure/azure-sdk-for-rust/releases).
There are other updates to other crates as well, such as Arrow Flight now being at 25.0.0, though I haven't really looked into it, I feel that we should do our best to keep up to date. This could all be done in another PR imo.

Reply:
Ah @KovacicAi I see those oauth2 changes [here](https://github.com/Azure/azure-sdk-for-rust/pull/9980) and [here](https://github.com/Azure/azure-sdk-for-rust/pull/997). These would be good things to know when transitioning to 0.6.0.